core/src/bson_type_info.rs

```diff
@@ -483,6 +483,27 @@ impl BsonTypeInfo {
         simple_type_info: SimpleTypeInfo::default(),
     };
 
+    // Datatype needed for backwards compatibility
+    pub const DATE_OLD: BsonTypeInfo = BsonTypeInfo {
+        type_name: "date",
+        sql_type: SqlDataType::EXT_TIMESTAMP,
+        non_concise_type: SqlDataType::EXT_TIMESTAMP,
```

Comment:
I wasn't sure about this -- ODBC 3's SQL_DATETIME overwrites the code for ODBC 2's SQL_DATE (both use 9). So should this be SQL_DATETIME because it's supposed to be SQL_DATE?

Reply:
Uhm, SQL_DATETIME and SQL_DATE are identifiers. I think SQL_DATETIME is still the correct one to use.
The concise type is just for date type times and intervals.
It basically just flags the types as being a general date time type. Then, using the sub code, one can tell which kind of date time type it is (e. i 3 for timestamp).

> For interval and datetime data types, the SQL_DATA_TYPE (== concise type) field in the result set will return SQL_INTERVAL or SQL_DATETIME, and the SQL_DATETIME_SUB field will return the subcode for the specific interval or datetime data type